diff --git a/llvm/include/llvm/TargetParser/Triple.h b/llvm/include/llvm/TargetParser/Triple.h
--- a/llvm/include/llvm/TargetParser/Triple.h
+++ b/llvm/include/llvm/TargetParser/Triple.h
@@ -265,6 +265,7 @@ public:
     CODE16,
     EABI,
     EABIHF,
     Android,
+    Ardos,
     Musl,
     MuslABIN32,
     MuslABI64,
diff --git a/llvm/lib/TargetParser/Triple.cpp b/llvm/lib/TargetParser/Triple.cpp
--- a/llvm/lib/TargetParser/Triple.cpp
+++ b/llvm/lib/TargetParser/Triple.cpp
@@ -331,6 +331,7 @@
   switch (Kind) {
   case UnknownEnvironment: return "unknown";
   case Android: return "android";
+  case Ardos: return "ardos";
   case CODE16: return "code16";
   case CoreCLR: return "coreclr";
   case Cygnus: return "cygnus";
@@ -739,6 +740,7 @@
       .StartsWith("gnut64", Triple::GNUT64)
       .StartsWith("gnu", Triple::GNU)
       .StartsWith("android", Triple::Android)
+      .StartsWith("ardos", Triple::Ardos)
       .StartsWith("muslabin32", Triple::MuslABIN32)
       .StartsWith("muslabi64", Triple::MuslABI64)
       .StartsWith("musleabihf", Triple::MuslEABIHF)
