diff -r -u a/config.sub b/config.sub
--- a/config.sub
+++ b/config.sub
@@ -2099,6 +2100,7 @@
 	| qnx* \
 	| rdos* \
 	| redox* \
+	| ardos* \
 	| rhapsody* \
 	| riscix* \
 	| riscos* \
@@ -2216,7 +2218,7 @@
 case $kernel-$os-$obj in
 	linux-gnu*- | linux-android*- | linux-dietlibc*- | linux-llvm*- \
 		    | linux-mlibc*- | linux-musl*- | linux-newlib*- \
+		    | linux-relibc*- | linux-uclibc*- | linux-ohos*- | linux-ardos*- )
-		    | linux-relibc*- | linux-uclibc*- | linux-ohos*- )
 		;;
 	uclinux-uclibc*- | uclinux-gnu*- )
 		;;
